web/src/server_events.js

```diff
@@ -264,6 +265,14 @@ export function force_get_events() {
 export function home_view_loaded() {
     waiting_on_homeview_load = false;
     get_events_success([]);
+
+    // If user has the all messages (aka home) view as their default
+    // and current view, call select_id on current selected message
+    // again to trigger mark_read event for selected message.
+    if (!recent_topics_util.is_visible() && message_lists.current === message_lists.home) {
+        const current_selected_message = message_lists.home.selected_id();
+        message_lists.home.select_id(current_selected_message);
+    }
```

Comment:
What's the thinking behind putting this hook here that will mark things as read? As I understand it, `home_view_loaded` is about our having the data to fully render the home view all the way to the present ... not about some minimal level required to show it at all.

Reply:
Hmm... yeah, it's probably better to add this hook in `message_fetch.js`. Before where we call `home_view_loaded` perhaps?

The idea is that now that we're fixing the bug noted above by passing `mark_read: false` to the home message list when we initially set the selected message, we need another `select_id` call for the home message list (if it's our current view on loading) to potentially mark the selected message as read (depending on the unread state and the user's settings).

Reply:
Moved this to `message_fetch.initialize` as part of the `load_more` function.